bika/lims/utils/analysisrequest.py

```diff
@@ -475,3 +488,33 @@ def fields_to_dict(obj, skip_fields=None):
             continue
         data[field_name] = field.get(obj)
     return data
+
+
+def resolve_rejection_reasons(values):
+    """Resolves the rejection reasons from the submitted values to the format
+    supported by Sample's Rejection Reason field
+    """
+    rejection_reasons = values.get("RejectionReasons")
+    if not rejection_reasons:
+        return []
+
+    # Predefined reasons selected?
+    selected = rejection_reasons[0] or {}
+    if selected.get("checkbox") == "on":
+        selected = selected.get("multiselection") or []
+    else:
+        selected = []
+
+    # Other reasons set?
+    other = values.get("RejectionReasons.textfield")
+    if other:
+        other = other[0] or {}
+        other = other.get("other", "")
+    else:
+        other = ""
+
+    # If neither selected nor other reasons are set, return empty
+    if any([selected, other]):
+        return {"selected": selected, "other": other}
```

Comment:
Please return a `list` type here as well. At the moment this is implicitly handled by the Records Field: https://github.com/raphael-ritz/Products.ATExtensions/blob/master/Products/ATExtensions/field/records.py#L106

Reply:
Well spotted, thanks!